fastlane/Fastfile

```diff
@@ -382,7 +382,7 @@ platform :ios do
     after_confirming_push(push_merge_branch: true) do
       trigger_beta_build(branch_to_build: release_branch_name)
 
-      copy_branch_protection(repository: GHHELPER_REPO, from_branch: DEFAULT_BRANCH, to_branch: release_branch_name)
+      set_branch_protection(repository: GHHELPER_REPO, branch: release_branch_name)
```

Comment:
@mokagio by copying `trunk` protection rules we aren't able to push changes to the release branch (such as bumping version).

We're reverting that to the previous way to avoid this issue in the next code freeze.

The options that are enabled now and that weren't before are the following:

![Screenshot 2024-06-25 at 13 17 17](https://github.com/Automattic/pocket-casts-ios/assets/7040243/1192aa18-bdc8-4387-9d81-68d6cc786599)

![Screenshot 2024-06-25 at 13 17 36](https://github.com/Automattic/pocket-casts-ios/assets/7040243/a2341959-e657-4b2e-be74-141f2bcc6e4b)